pynamodb/models.py

```diff
@@ -648,6 +648,9 @@ def query(cls,
                 limit -= 1
             yield cls.from_raw_data(item)
 
+        if limit is None and page_size is not None:
```

Comment:
Please see my changes. I want to make sure that we don't scan more than the `page_size`. Incase when the result size is > 1MB, we keep scanning until we reach a total amount of `page_size` scan.